packages/server/modules/workspacesCore/domain/events.ts

```diff
@@ -8,7 +8,8 @@ export const WorkspaceEvents = {
   Created: `${workspaceEventPrefix}created`,
   Updated: `${workspaceEventPrefix}updated`,
   RoleDeleted: `${workspaceEventPrefix}role-deleted`,
-  RoleUpdated: `${workspaceEventPrefix}role-updated`
+  RoleUpdated: `${workspaceEventPrefix}role-updated`,
+  Discovered: `${workspaceEventPrefix}discovered`
```

Comment:
What's discovered mean? If it means "user joined discoverable workspace" then name is a bit misleading - the "discovery" already happened when the user noticed these workspaces in the banner. `JoinDiscoverable` or just `Join` might be a better name